File: src/main/g8/$service__packaged$/src/main/$java__packaged$/$servicePackage__packaged$/hello/HelloServiceImpl.java
package com.isuwang.soa.hello;

import com.isuwang.dapeng.core.SoaException;
import com.isuwang.soa.hello.domain.Hello;
import com.isuwang.soa.hello.service.HelloService;

public class HelloServiceImpl implements HelloService{
    @Override
    public String sayHello(Hello hello) throws SoaException {
        return "hello: " + hello.toString();
    }

    @Override
    public void test(String content) throws SoaException {
        System.out.println(" hello world: " + content);
    }
}

File: src/main/g8/$service__packaged$/src/main/$scala__packaged$/$servicePackage__packaged$/scala/hello/HelloServiceImpl.scala
package com.isuwang.soa.scala.hello

import com.isuwang.soa.hello.scala.domain.Hello
import com.isuwang.soa.hello.scala.service.HelloService
import javax.annotation.Resource
import javax.sql.DataSource

import com.isuwang.soa.hello.scala.service.HelloService
import wangzx.scala_commons.sql._
import entity.Company


class HelloServiceImpl extends HelloService {

  @Resource(name = "crm_dataSource")
  var dataSource: DataSource = _

  /**
    *
    **/
  override def sayHello(hello: Hello): String = {
    val company = dataSource.row[Company](sql" select * from companys where id = 119501")
    println(s" companyInfo: \${company}")
    company match {
      case Some(c) => c.toString
      case None => ""
    }
  }

  /**
    *
    **/
  override def test(content: String): Unit = {
    println(s" received content: \${content}")
  }
}
